From ed5de4d550f1de218e57ae555fda51829fa511a2 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Mon, 19 Aug 2024 20:53:27 +0200
Subject: [PATCH] Revert ChatGPT recommendations

---
 pyfileindex/pyfileindex.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/pyfileindex/pyfileindex.py b/pyfileindex/pyfileindex.py
index 46c0b8f..b298d0b 100644
--- a/pyfileindex/pyfileindex.py
+++ b/pyfileindex/pyfileindex.py
@@ -82,7 +82,7 @@ def open(self, path: str) -> "PyFileIndex":
                 filter_function=self._filter_function,
                 debug=self._debug,
                 df=self._df[
-                    pandas.Series(
+                    np.array(
                         [
                             p.replace("\\", "/").contains(abs_path_unix)
                             for p in self._df["path"].values
@@ -192,7 +192,7 @@ def _get_changes_quick(self) -> tuple:
             tuple: pandas.DataFrame with new entries, list of changed files, and list of deleted paths
         """
         path_exists_bool_lst = [os.path.exists(p) for p in self._df.path.values]
-        path_deleted_lst = self._df[~pandas.Series(path_exists_bool_lst)].path.values
+        path_deleted_lst = self._df[~np.array(path_exists_bool_lst)].path.values
         df_exists = self._df[path_exists_bool_lst]
         stat_lst = [os.stat(p) for p in df_exists.path.values]
         st_mtime = [s.st_mtime for s in stat_lst]
