From 43d36eb67d1ef188a0fd38fbc86da69b9d956e78 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan=20Jan=C3=9Fen?= <janssen@mpie.de>
Date: Sat, 13 May 2023 20:15:11 -0700
Subject: [PATCH] update before open

---
 pyfileindex/pyfileindex.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/pyfileindex/pyfileindex.py b/pyfileindex/pyfileindex.py
index 8796a30..6cbaa9a 100644
--- a/pyfileindex/pyfileindex.py
+++ b/pyfileindex/pyfileindex.py
@@ -53,6 +53,7 @@ def open(self, path):
         """
         abs_path = os.path.abspath(os.path.expanduser(os.path.join(self._path, path)))
         self._check_if_path_exists(path=abs_path)
+        self.update()
         if abs_path == self._path:
             return self
         elif os.path.commonpath([abs_path, self._path]) == self._path:
